x-pack/plugins/alerting/server/alerts_client/alerts_client.test.ts

```diff
@@ -81,406 +82,310 @@ const mockSetContext = jest.fn();
+                  uuid: 'def',
                 },
-                maintenance_window_ids: [],
-                rule: {
-                  category: 'My test rule',
-                  consumer: 'bar',
-                  execution: {
-                    uuid: '5f6aa57d-3e22-484e-bae8-cbed868f4d28',
-                  },
-                  name: 'rule-name',
-                  parameters: {
-                    bar: true,
-                  },
-                  producer: 'alerts',
-                  revision: 0,
-                  rule_type_id: 'test.rule-type',
-                  tags: ['rule-', '-tags'],
-                  uuid: '1',
+              }),
+            },
+            recovered: {
+              '3': new Alert('3', {
+                state: { foo: false },
+                meta: {
+                  flapping: false,
+                  flappingHistory: [true, false, false],
+                  uuid: 'xyz',
                 },
-                start: date,
-                status: 'active',
-                time_range: {
-                  gte: date,
+              }),
+            },
+          }));
+          const spy = jest
+            .spyOn(LegacyAlertsClientModule, 'LegacyAlertsClient')
+            .mockImplementation(() => mockLegacyAlertsClient);
+
+          const alertsClient = new AlertsClient(alertsClientParams);
+
+          const opts = {
+            maxAlerts,
+            ruleLabel: `test: rule-name`,
+            flappingSettings: DEFAULT_FLAPPING_SETTINGS,
+            activeAlertsFromState: {},
+            recoveredAlertsFromState: {},
+          };
+          await alertsClient.initializeExecution(opts);
+          expect(mockLegacyAlertsClient.initializeExecution).toHaveBeenCalledWith(opts);
+
+          expect(clusterClient.search).toHaveBeenCalledWith({
+            body: {
+              query: {
+                bool: {
+                  filter: [
+                    { term: { 'kibana.alert.rule.uuid': '1' } },
+                    { terms: { 'kibana.alert.uuid': ['abc', 'def', 'xyz'] } },
+                  ],
                 },
-                uuid: uuid1,
-                workflow_status: 'open',
               },
-              space_ids: ['default'],
-              version: '8.9.0',
-            },
-            tags: ['rule-', '-tags'],
-          },
-          { index: { _id: uuid2 } },
-          // new alert doc
-          {
-            '@timestamp': date,
-            event: {
-              action: 'open',
-              kind: 'signal',
+              seq_no_primary_term: true,
+              size: 3,
             },
-            kibana: {
-              alert: {
-                action_group: 'default',
-                duration: {
-                  us: '0',
-                },
-                flapping: false,
-                flapping_history: [true],
-                instance: {
-                  id: '2',
+            index: useDataStreamForAlerts
```

Comment:
There are a lot of snippets like this in the tests - inline testing of whether data streams are being used.

I don't feel like I had a good "view" of this while hacking away, fixing these.  Perhaps a lot of these can be re-org'd somehome into some simple functions ... as a later enhancement.